testing/unittests/esdltests.cpp

```diff
@@ -1734,6 +1736,147 @@ constexpr const char * result = R"!!(<soap:Envelope xmlns:soap="http://schemas.x
+        static const char* levelsResult8 = R"!!!({"level-default": "", "level-min": "", "level-normal": "", "level-1": "", "level-2": "", "level-3": "", "level-4": "", "level-5": "", "level-6": "", "level-7": "", "level-8": ""})!!!";
+        static const char* levelsResult9 = R"!!!({"level-default": "", "level-min": "", "level-normal": "", "level-1": "", "level-2": "", "level-3": "", "level-4": "", "level-5": "", "level-6": "", "level-7": "", "level-8": "", "level-9": ""})!!!";
+        static const char* levelsResult10 = R"!!!({"level-default": "", "level-min": "", "level-normal": "", "level-max": "", "level-1": "", "level-2": "", "level-3": "", "level-4": "", "level-5": "", "level-6": "", "level-7": "", "level-8": "", "level-9": "", "level-10": ""})!!!";
+        static constexpr const char * typesScript = R"!!(<es:CustomRequestTransform xmlns:es="urn:hpcc:esdl:script" target="Request">
+          <es:tx-summary name="type-default" value="0"/>
+          <es:tx-summary name="type-text" value="0" type="text"/>
+          <es:tx-summary name="type-signed-min" value="-9223372036854775808"/>
+          <es:tx-summary name="type-signed-max" value="9223372036854775807"/>
+          <es:tx-summary name="type-unsigned-min" value="0" type="unsigned"/>
+          <es:tx-summary name="type-unsigned-max" value="18446744073709551615" type="unsigned"/>
+          <es:tx-summary name="type-decimal-min" value="-1.79769e+308" type="decimal"/>
+          <es:tx-summary name="type-decimal-max" value="1.79769e+308" type="decimal"/>
+          <es:set-value target="types-all" select="getTxSummary('json')"/>
+        </es:CustomRequestTransform>
+        )!!";
+        static const char* typesResultAll = R"!!!({"type-default": "0", "type-text": "0", "type-signed-min": -9223372036854775808, "type-signed-max": 9223372036854775807, "type-unsigned-min": 0, "type-unsigned-max": 18446744073709551615, "type-decimal-min": -1.79769e+308, "type-decimal-max": 1.79769e+308})!!!";
+        static constexpr const char * timersScript = R"!!(<es:CustomRequestTransform xmlns:es="urn:hpcc:esdl:script" target="Request">
+          <es:tx-summary-timer name="scalar-default">
+            <es:delay ms="2"/>
+          </es:tx-summary-timer>
+          <es:tx-summary-timer name="scalar-append" mode="append">
+            <es:delay ms="2"/>
+          </es:tx-summary-timer>
+          <es:tx-summary-timer name="scalar-set" mode="set">
+            <es:delay/>
+          </es:tx-summary-timer>
+          <es:tx-summary-timer name="accumulating" mode="accumulate">
+            <es:delay ms="5"/>
+          </es:tx-summary-timer>
+          <es:tx-summary-timer name="accumulating" mode="accumulate">
+            <es:delay ms="10"/>
+          </es:tx-summary-timer>
+          <es:set-value target="timers-all" select="getTxSummary('json')"/>
```

Comment:
This is failing because the `delay` operation property was changed  from `ms` to `millis`. The failure was not detected because areMatchingPTrees, used to compare the summary results, never compares element values; it only compares element names, attribute names, and attribute values. A match is erroneously confirmed because the element names match.

Reply:
Will use nlohmann::json, used by the jwtSecurity security manager, to do value comparisons.